File: frontend/src/types/generator.ts
export interface GenerateRequest {
  start_date: string;
  end_date: string;
  num_rows: number;
  seed: number;
  inject_r1: boolean;
  inject_r2: boolean;
  inject_r3: boolean;
  inject_r4: boolean;
  inject_r5: boolean;
  inject_r6: boolean;
}

export interface GenerateStats {
  total_rows: number;
  amount_mean: number;
  round_amounts: number;
}

export interface GenerateResponse {
  stats: GenerateStats;
  preview: Record<string, unknown>[];
  download_token: string;
}

File: backend/schemas/generator.py
from pydantic import BaseModel, Field


class GenerateRequest(BaseModel):
    start_date: str
    end_date: str
    num_rows: int = Field(default=50000, ge=1000, le=100000)
    seed: int = Field(default=42, ge=0, le=9999)
    inject_r1: bool = True
    inject_r2: bool = True
    inject_r3: bool = True
    inject_r4: bool = True
    inject_r5: bool = True
    inject_r6: bool = True


class GenerateStats(BaseModel):
    total_rows: int
    amount_mean: float
    round_amounts: int


class GenerateResponse(BaseModel):
    stats: GenerateStats
    preview: list[dict]
    download_token: str
